extensions/amp-access-laterpay/0.1/amp-access-laterpay.js

```diff
@@ -28,27 +87,371 @@ export class LaterpayVendor {
    * @param {!AccessService} accessService
    */
   constructor(accessService) {
-    /** @private @const */
+    /** @const @private {!AccessService} */
     this.accessService_ = accessService;
-    // TODO: implement
+
+    /** @const @private {!Window} */
+    this.win_ = this.accessService_.win;
+
+    /** @const @private {!Document} */
+    this.doc_ = this.win_.document;
+
+    /** @private @const {!Viewport} */
+    this.viewport_ = viewportForDoc(this.win_.document);
+
+    /** @const @private {!LaterpayConfigDef} */
+    this.laterpayConfig_ = this.accessService_.getAdapterConfig();
+
+    /** @private {?PurchaseConfigDef} */
+    this.purchaseConfig_ = null;
+
+    /** @private {?Function} */
+    this.purchaseButtonListener_ = null;
+
+    /** @private {?Function} */
+    this.alreadyPurchasedListener_ = null;
+
+    /** @const @private {!Array<Event>} */
+    this.purchaseOptionListeners_ = [];
+
+    /** @private {!boolean} */
+    this.containerEmpty_ = true;
+
+    /** @private {?Node} */
+    this.selectedPurchaseOption_ = null;
+
+    /** @private {?Node} */
+    this.purchaseButton_ = null;
+
+    const configUrl = (
+      (getMode().localDev || getMode().development) &&
+      this.laterpayConfig_.configUrl
+    ) ? this.laterpayConfig_.configUrl
+      : CONFIG_URL;
+
+    /** @private {string} */
+    this.currentLocale_ = this.laterpayConfig_.locale || 'en';
+
+    /** @private {Object} */
+    this.i18n_ = Object.assign(DEFAULT_MESSAGES,
+                  this.laterpayConfig_.localeMessages || {});
+
+    /** @private {string} */
+    this.purchaseConfigBaseUrl_ = configUrl + CONFIG_BASE_PATH;
+    const articleId = this.laterpayConfig_.articleId;
+    if (articleId) {
+      this.purchaseConfigBaseUrl_ += '&article_id=' + encodeURIComponent(articleId);
+    }
+
+    /** @const @private {!Timer} */
+    this.timer_ = timerFor(this.win_);
+
+    /** @const @private {!Vsync} */
+    this.vsync_ = vsyncFor(this.win_);
+
+    /** @const @private {!Xhr} */
+    this.xhr_ = xhrFor(this.win_);
+
+    installStyles(this.win_.document, CSS, () => {}, false, TAG);
```

Comment:
There shouldn't be multiple instances of this extension, no, so I'd rather add a check for that.

Given that the instance of this class is already [created at the end of this file](https://github.com/laterpay/amphtml/blob/8be201046ac6b5b135dfcdbcc549b82f9d47cf67/extensions/amp-access-laterpay/0.1/amp-access-laterpay.js#L461) when the vendor is registered, I'm wondering it it's already the case?